addons/event/models/event_registration.py

```diff
@@ -149,6 +149,8 @@ def _synchronize_partner_values(self, partner, fnames=None):
 
     @api.model_create_multi
     def create(self, vals_list):
+        if self.env.user._is_public():
```

Comment:
This should not be done at that level. Caller may sudo the call to create if we are sure access is granted beforehand ...